plutus-contract/src/Plutus/Contract/Request.hs

```diff
@@ -151,6 +150,36 @@ waitNSlots n = do
   c <- currentSlot
   awaitSlot $ c + fromIntegral n
 
+-- | Wait until the given time.
+awaitTime ::
+    forall w s e.
+    ( AsContractError e
+    )
+    => POSIXTime
+    -> Contract w s e POSIXTime
+awaitTime s = pabReq (AwaitTimeReq s) E._AwaitTimeResp
+
+-- | Get the current time
+currentTime ::
+    forall w s e.
+    ( AsContractError e
+    )
+    => Contract w s e POSIXTime
+currentTime = pabReq CurrentTimeReq E._CurrentTimeResp
+
+-- | Wait for a number of seconds to pass.
+--
+-- Note: Currently, if n < length of a slot, then 'waitNSeconds' has no effect.
+waitNSeconds ::
+  forall w s e.
+  ( AsContractError e
+  )
+  => Natural
```

Comment:
Hmm. Maybe we should define a `DiffSeconds` type so we have something more descriptive to take as an argument for things like this?

Reply:
Should we reuse `NominalDiffTime` from `Data.Time.Clock`?

Reply:
I guess not, if it might be used onchain...

Reply:
Exactly.